apps/arweave/src/ar_storage.erl

```diff
@@ -299,28 +299,30 @@ put_tx_confirmation_data(B) ->
 %% @doc Return {BlockHeight, BlockHash} belonging to the block where
 %% the given transaction was included.
 get_tx_confirmation_data(TXID) ->
-	case ar_kv:get(tx_confirmation_db, TXID) of
-		{ok, Binary} ->
-			{ok, binary_to_term(Binary)};
-		not_found ->
-			{ok, Config} = application:get_env(arweave, config),
-			case lists:member(arql, Config#config.disable) of
-				true ->
-					not_found;
-				_ ->
-					case catch ar_arql_db:select_block_by_tx_id(ar_util:encode(TXID)) of
-						{ok, #{
-							height := Height,
-							indep_hash := EncodedIndepHash
-						}} ->
-							{ok, {Height, ar_util:decode(EncodedIndepHash)}};
-						not_found ->
-							not_found;
-						{'EXIT', {timeout, {gen_server, call, [ar_arql_db, _]}}} ->
-							{error, timeout}
-					end
-			end
-	end.
+    case ar_kv:get(tx_confirmation_db, TXID) of
+        {ok, Binary} ->
+            {ok, binary_to_term(Binary)};
+        not_found ->
+            {ok, Config} = application:get_env(arweave, config),
+            case lists:member(arql, Config#config.disable) of
+                true ->
+                    {error, not_found};
+                false ->
+                    try ar_arql_db:select_block_by_tx_id(ar_util:encode(TXID)) of
+                        {ok, #{
+                            height := Height,
+                            indep_hash := EncodedIndepHash
+                        }} ->
+                            {ok, {Height, ar_util:decode(EncodedIndepHash)}};
+                        _ ->
+                            {error, not_found}
+                    catch
+                        _Class:_Exception ->
```

Comment:
looks like we lost the `{error, timeout}` return

Reply:
maybe take a look at the file now if it makes sense, I return error, timeout from the catch clause

Reply:
I think the issue is the formatting of returns. Previously only when we caught `{'EXIT', {timeout, {gen_server, call, [ar_arql_db, _]}}}` did we return `{error, timeout}` - now any error will trigger an `{error, timeout}`. I'm not sure if it will make a difference, but I'm a little wary of functionality changes in a large formatting PR.

If there aren't any callers of `get_tx_confirmation_data` that handle `{error, timeout}`, then it's fine and we can go with your earlier iteration of `{error, Exception}` (or whatever)

but if the callers of `get_tx_confirmation_data` handle `{error, timeout}` or handle `{error, SOMETHING}`, then this diff might break those callers handling logic.

Reply:
I don't want any functional change, I'm aware that enforcing single case statement may be mutually exclusive with that goal. But let's look at this:

```sh
> git grep get_tx_confirmation_data                                           
apps/arweave/src/ar_header_sync.erl:250:        case ar_storage:get_tx_confirmation_data(TXID) of
apps/arweave/src/ar_http_iface_middleware.erl:1543:                                     case ar_storage:get_tx_confirmation_data(TXID) of
apps/arweave/src/ar_storage.erl:10:             read_tx/1, read_tx_data/1, update_confirmation_index/1, get_tx_confirmation_data/1,
apps/arweave/src/ar_storage.erl:301:get_tx_confirmation_data(TXID) ->
apps/arweave/test/ar_tx_blacklist_tests.erl:324:                        ?assertMatch({ok, {_, _}}, ar_storage:get_tx_confirmation_data(TXID))
apps/arweave/test/ar_tx_blacklist_tests.erl:351:                        ?assertMatch({ok, {_, _}}, ar_storage:get_tx_confirmation_data(TXID)
```

Excluding tests, we have callers in ar_http_iface_middleware.erl and ar_header_sync.erl

Looking at ar_http_iface_middleware.erl
it pattern matches on

```erl
{ok, {Height, BH}} ->
.... elided ...
						not_found ->
							{404, #{}, <<"Not Found.">>, Req};
						{error, not_found} ->
							{404, #{}, <<"Not Found.">>, Req};
						{error, timeout} ->
							{503, #{}, <<"ArQL unavailable.">>, Req}
```

and ar_header_sync.erl

which pattern matches on

```erl
		{ok, {Height, _BH}} ->
.....
		not_found ->
			ar_events:send(tx, {ready_for_unblacklisting, TXID}),
			{noreply, State};
		{error, Reason} ->
			?LOG_WARNING([{event, failed_to_read_tx_confirmation_index},
					{error, io_lib:format("~p", [Reason])}]),
			{noreply, State}
```

the former one only matches timeout errors, while the latter one matches any error. So any other error, Reason that isn't timeout would not fly with ar_http_iface_middleware but would be ok in ar_header_sync.

I can write up a ticket to make it consistent and allow any error reason to be sent and received, because surely timeout is only one of meany reasons.

Reply:
Given what the callers are doing, and what the behavior was before, it looks like what you have now is good. The only change in functionality is now it will return {error, timeout} for some scenarios where it crashed outright before. I think that change in functionality is okay. (assuming I understand the whole `{'EXIT', {...` match which I'm not sure i do)